tests/suites/test_suite_hkdf.function

```diff
@@ -89,7 +89,7 @@ void test_hkdf_expand( int md_alg, char *hex_info_string,
     ret = mbedtls_hkdf_expand( md, prk, prk_len, info, info_len,
                                output_okm, OKM_LEN );
     TEST_ASSERT( ret == 0 );
-    TEST_ASSERT( !memcmp( output_okm, okm, okm_len ) );
+    ASSERT_COMPARE( output_okm, okm_len, okm, okm_len );
```

Comment:
Pre-existing and not clear if in scope: I really don't understand why we have `OKM_LEN` here, much less why we use it instead of `okm_len` in the call to `hkdf_expand()` above. Unless I'm missing something, this test should be improved by getting rid of `OKM_LEN` and also using `data_t` where applicable.

Reply:
Filed as https://github.com/ARMmbed/mbedtls/issues/3683

Reply:
ok thanks for filling the issue.